vendors/nordic/boards/nrf52840-dk/ports/common_io/config/iot_gpio_config.h

```diff
@@ -0,0 +1,55 @@
+/*
```

Comment:
This file could reside in the application space where other config files are today under aws_{demos,tests}/config_files

Reply:
Thank you, this is a great improvement and aligns with similar `*config.h` files we already have. I'll relocate this file as you suggested.

My only apprehension was that a similar file `iot_spi_config.h` resides in that same directory. As SPI is outside the scope of this PR, I'll create a separate PR to deal with that.

Reply:
Thank you. I added  `iot_gpio_config_defaults.h` in place of `iot_gpio_config.h` and moved `iot_gpio_config.h` to user config directory alongside other existing user config files.